pkg/mappings.bzl

```diff
@@ -66,19 +66,40 @@ strip_prefix = struct(
     from_root = _sp_from_root,
 )
 
+def pkg_attributes(mode = "0444", user = None, group = None, **kwargs):
+    """Format attributes for use in package mapping rules.
+
+    Args:
+      mode: string: UNIXy octal permissions, as a string.  Defaults to being read-only ("0444")
+      user: string: Filesystem owning user.
+      group: string: Filesystem owning group.
+      **kwargs: any other desired attributes.
+
+    Not providing any of "user", or "group" will result in the package builder
```

Comment:
Why is mode defaulted, but the rest not?  It might be valid to say
pkg_attributes(user='man') and let the builder default the mode.

Reply:
Thanks for taking a look

Mostly just supposition so far:

- The "sane" default for package managers that produce installable package is to set user/group to an administrative user.
- If you're not producing something that's installed, user/group generally do not matter, as they get replaced with other things.

That being said, you got me thinking again about the default for `mode`.  Virtually all source archives I've encountered have files at UNIX 0644 and directories at 0755.  If you explicitly want to tune permissions to be read only for everyone, then you can do whatever you want.

So, I'm thinking to do the following:

- Set the default attributes to the empty string (or empty json, depending on what works)
- Set default mode for pkg_files to `0644`
- Within each rule, establish defaults like the ones specified in rules now, then overlay the decoded input on top of it.

I think the behavior is easy enough to understand, but it deviates from most things I've seen in bazel so far (attribute always overrides default).  I don't think this will be a problem. WDYT?

Reply:
I went for a "set a sane default for mode and leave everything else alone" option, with updated documentation and tests.  PTAL.

Reply:
I would think about what happens when you do this
```
cc_binary(name='x')
pkg_file(name='files', srcs=[':x', ':y'], ...)
pkg_rpm(name='pkg', srcs=[':files'])
```
The least surprising thing is if `x` is mode 0755 at the end.  I think what you have now will allow that to happen.

Reply:
That's a good point, however I don't think we're quite there.  We'd have to do some introspection magic in `pkg_files` to identify the nature of the dependent rules.  Providers?  Rule names?  This will need to be investigated.  At first brush, this feels potentially error-prone.

I feel that this can be postponed to a later task.  I'll make a ticket for it.